### PR TITLE
Add support for anything renderable as children rather than just strings.

### PR DESCRIPTION
**Pull request by kchunterdeluxe**
Hi all,

We recently upgraded to v5.0.3 and ran across the following error:
image

The error is due to only children of type `string` being supported rather than anything renderable. This PR contains a failing test with a example of using JS expressions within children.

From React docs https://facebook.github.io/react/docs/jsx-in-depth.html#children-in-jsx

> In JSX expressions that contain both an opening tag and a closing tag, the content between those tags is passed as a special prop: props.children. There are several different ways to pass children [...]

Where children can typically be anything that can be rendered: numbers, strings, elements or an array (or fragment) containing these types (from [PropTypes.children](https://facebook.github.io/react/docs/typechecking-with-proptypes.html#proptypes))

With the current API, I would expect `children` of these various types to be supported. However for example, `<title>` requires the `children` to be `string`s.

To summarize by examples:

```js
const title = 'The Title';

// Works
<Helmet>
  <title>Title: The Title</title>
</Helmet>

// Works
<Helmet>
  <title>{'Title: The Title'}</title>
</Helmet>

// Works
<Helmet>
  <title>{'Title: ' + title}</title>
</Helmet>

// Doesn't Work
<Helmet>
  <title>Title: {title}</title>
</Helmet>
```

### CODE REVIEW

**Comment by cwelch5** [comment]
@kchunterdeluxe Could you sign the CLA please?  So we can give you credit for the unit test.  Thanks!

**Comment by kchunterdeluxe** [comment]
@cwelch5, CLA signed
